### ISSUE TITLE
New license request: CORE [SPDX-Online-Tools]

### ISSUE THREAD

**Issue by rastislavcore** [spdx/license-list-XML#2607]
**1.** License Name: CORE License
**2.** Short identifier: CORE
**3.** License Author or steward: Blockchain Hub
**4.** Comments: The CORE License promotes open-source software principles by allowing unrestricted use, modification, and distribution of software, provided that all distributions of the software in source code form, including modifications, remain publicly accessible. It aligns with SPDX principles by being publicly available, widely applicable, and facilitating software freedom and compliance management.
**5.** License Request Url: http://tools.spdx.org/app/license_requests/406
**6.** URL(s): https://github.com/bchainhub/core-license/blob/master/LICENSE
**7.** OSI Status: Not Submitted
**8.** Example Projects: 
**8.** License Text Diff: https://github.com/spdx/licenseRequestImages/blob/master/d05e6395-2d6d-43c3-bb28-c37f91fbc9aa.png

**Note:**
The license closely matched with the following license ID(s): MIT

**Comment by rastislavcore** [comment]
Certainly! Here are several FOSS projects actively using this license:

- https://github.com/DataLayerHost/txms-server
- https://github.com/bchainhub/txms.js
- https://github.com/bchainhub/remark-fediverse-user
- https://github.com/core-laboratories/XmlGoaml
- https://github.com/catchthatrabbit/pool-frontend
- https://github.com/core-coin/pigeon

These projects demonstrate the license's utility across various open-source initiatives.

**Comment by richardfontana** [comment]
It looks like all but possibly [pigeon](https://github.com/core-coin/pigeon) are connected to the issue opener. 

Regarding the assertion that these are FOSS/open source projects, that is an interesting question here (relevant because of the SPDX license inclusion criteria). The license is based on the MIT license but replaces the notice preservation paragraph with this clause:

> All distribution of the Covered Software in Source Code Form, including any
Modifications and/or Contributions must be disclosed and publicly available.

This uses the capitalized terms "Covered Software", "Source Code Form", "Modifications" and "Contributions" without defining them (I wonder whether the license was influenced by certain well known open source licenses that do have these as defined terms?). 

Anyway, the truly noteworthy feature here is (as I read this) an attempt to require that all distribution of any sort be "public", i.e. presumably prohibiting merely distributing to one person. I'd assert that this fails traditional community standards for open source (which I would argue must *generally* permit you to choose your immediate distributees). Perhaps it's open to debate, but that debate hasn't been had anywhere as far as I am aware.

**Comment by karsten-klein** [comment]
**{metæffekt} Universe**
canonical name: [CORE License](https://github.com/org-metaeffekt/metaeffekt-universe/blob/main/src/main/resources/ae-universe/[c]/README.md#user-content-CORE-License:~:text=CORE%20License)
short name: CORE-License
markers: No Warranty Marker
category: CORE License
OSI status: _none_

**ScanCode**
matched id: [mit-0](https://github.com/nexB/scancode-toolkit/blob/develop/src/licensedcode/data/licenses/mit-0.LICENSE)

**Comment**
+1 to add due to differentiated obligation and induced risk.

**Comment by rastislavcore** [comment]
A relevant software reference that is unrelated to me as owner - author and developed independently by another author, whom I have no connection with, is [Blockchain Wallet Validator](https://github.com/sergical/blockchain-wallet-validator). This project provides a validation framework for blockchain wallet addresses across multiple networks. I did contribute to the project, to improve the quality.

Other software distributions are gradually emerging, and there is significant potential for wider adoption in the future. As the ecosystem evolves, discussions on license clarity, compliance with FOSS principles, and broader community adoption will become increasingly important.

**Comment by rastislavcore** [comment]
We have updated the README file to provide additional insights into the licensing details. You can review the changes here: [Core License Repository](https://github.com/bchainhub/core-license).

**Comment by jlovejoy** [comment]
While the rationale page at https://github.com/bchainhub/core-license/tree/main is a nice-to-have, it is not binding in terms of legal interpretation. It is the license text that stands and as noted, the use of 4 capitalized yet undefined terms makes the license difficult if not impossible to interpret as it suggests there is missing info or someone made a very basic drafting mistake of capitalizing terms without defining them. The SPDX license inclusion definitive factor D states, "The license has identifiable and stable text; it is not in the midst of drafting." and while this does not mention poor or incomplete drafting, I would interpret "stable" text as being able to expand to this concept.  

Legal drafting need to prioritize clarify first and foremost (this is coming from someone who spends a lot of time drafting and reviewing various forms of legal language, a task I do not consider anyone ever being a perfect expert at - feedback is always a gift in this task :)

**Comment by jlovejoy** [comment]
I would also add from more of my own open source license/community perspective: we have enough less-than-perfectly drafted licenses in the open source software ecosystem that have been around for years and we are stuck sort of stuck with them. As a community, we should endeavor to not make more/new licenses that introduce new/more lack of clarity. To me, that is another form of "license proliferation"!

**Comment by Pizza-Ria** [comment]
+1 to @richardfontana & @jlovejoy - disinclined to add.

**Comment by jlovejoy** [comment]
given we have had 3 people weigh in, I'm going to mark this as not accepted at this time. If the license gets revised and more uptake, we can always reconsider.
